packages/clay-core/src/vertical-bar/Resizer.tsx

```diff
@@ -0,0 +1,77 @@
+/**
+ * SPDX-FileCopyrightText: © 2023 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import React, {useContext} from 'react';
+
+import {VerticalBarContext} from './context';
+
+type Props = {
+	/**
+	 * Indicates whether the panel can be focused.
+	 */
+	nodeRef?: React.RefObject<HTMLDivElement>;
+};
+
+export function Resizer({nodeRef}: Props) {
+	const {onPanelWidthChange, position, resize} =
+		useContext(VerticalBarContext);
+
+	return (
+		<>
+			{resize ? (
+				<div
+					aria-orientation="horizontal"
+					className="c-horizontal-resizer"
+					onPointerDown={(event) => {
+						const positionLeft = position === 'left';
+
+						const startWidth = nodeRef?.current?.offsetWidth ?? 320;
+						const startXPos = event.pageX;
+
+						function onResizerMove(event: any) {
+							const delta = Math.abs(event.pageX - startXPos);
+
+							if (
+								(event.pageX >= startXPos && positionLeft) ||
+								(event.pageX < startXPos && !positionLeft)
+							) {
+								onPanelWidthChange(
+									Math.round(startWidth + delta)
+								);
+							} else if (
+								(event.pageX < startXPos && positionLeft) ||
+								(event.pageX >= startXPos && !positionLeft)
+							) {
+								onPanelWidthChange(
+									Math.round(startWidth - delta)
+								);
+							}
```

Comment:
We have to add minimum and maximum values to the width size, an implementation of clamp should take care of this, the minimum will probably have to be the default size.